python/ray/tune/integration/mlflow.py

```diff
@@ -311,7 +323,11 @@ def __init__(self, config: Dict, *args, **kwargs):
 
         # First see if experiment_id is passed in.
         experiment_id = mlflow_config.pop("experiment_id", None)
-        if experiment_id is None:
+        if experiment_id is None or self._mlflow.get_experiment(
+                experiment_id) is None:
+            logger.debug("Either no experiment_id is passed in, or the "
+                         "experiment with the given id does not exist. "
+                         "Checking experiment_name")
             # Check for name.
             experiment_name = mlflow_config.pop("experiment_name", None)
```

Comment:
nit: `config.pop(x) == config.pop(x, None)`

Reply:
Good point- fixed

Reply:
Apparently this isn't the case. `config.pop(x)` throws a KeyError if x is not in the dict.